levelZeroLib/src/levelZeroContext.cpp

```diff
@@ -328,7 +328,18 @@ JNIEXPORT jint JNICALL Java_uk_ac_manchester_tornado_drivers_spirv_levelzero_Lev
     env->SetIntField(javaCommandQueueDescriptor, field, commandQueueDesc.stype);
 
     field = env->GetFieldID(commanddescriptorClass, "pNext", "J");
-    env->SetLongField(javaCommandQueueDescriptor, field, reinterpret_cast<jlong>(commandQueueDesc.pNext));
+#ifdef _WIN32
+    jlong pNext = env->GetLongField(javaCommandQueueDescriptor, field);
+#else
+    jint pNext = env->GetLongField(javaCommandQueueDescriptor, field);
+#endif
```

Comment:
Thanks @stratika . We can probably avoid duplicates with a C-Macro to get the right value for the pointer.

Reply:
sure, I will try to fix that.

Reply:
lust updated the file, let me know what you think.